lib/std/with.nim

```diff
@@ -49,3 +52,14 @@ when isMainModule:
   with(f, setColor(2, 3, 4), setPosition(0.0, 1.0))
   echo f
 
+  f = Foo()
+  with f:
+    col = $(2, 3, 4)
+    pos = $(0.0, 1.0)
+    name = "bar"
+  echo f
+
+  let f2 = f.dup(setPosition(2.0, 3.0)).with:
+    col = $(3, 4, 5)
+  echo f2
```

Comment:
I think you misunderstood. What I meant was that since `sugar.dup` also uses `underscoredCall`, your changes to it should enable it to have syntax like this too:

```nim
let f2 = f.dup:
  setPosition(2.0, 3.0)
  col = $(3, 4, 5)
```

I asked if you could verify this to be the case and add tests for `dup()` to ensure it will keeps working, not adding special casing for `.dup.with` :P

Reply:
The following code  already works:
```nim
let f2 = f.dup:
  setPosition(2.0, 3.0)
  col = $(3, 4, 5)
```

Reply:
Then please add a test for it (in sugar module I persume), and revert the `.dup.with` change.